test/healthcheck_test.go

```diff
@@ -203,6 +203,45 @@ func TestHealthCheckWatchStateChange(t *testing.T) {
 	}
 }
 
+// If Watch returns Unimplemented, then the ClientConn should go into READY state.
+func TestHealthCheckHealthServerNotRegistered(t *testing.T) {
+	defer leakcheck.Check(t)
+	s := grpc.NewServer()
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatal("Failed to listen")
```

Comment:
I've fixed in my merged PR that the returned error should be included in the test error message. Essentially, `t.Fatalf("failed to listen due to err: %v", err)`.